### PR TITLE
fix: WFO1000 Property does not configure the code serialization for i…

### PR DESCRIPTION
**Pull request by gerhardol**
…ts property content

## Proposed changes

Content serialization must be declared with .NET9, error WFO1000 
Similar change is needed for conemu, icsharp submodules
An alternative is to supress the warning for .net9

I took .net9 for a quick spin, this was the biggest required change. A few other suppression needed due to deprecations.
The app starts. With the favorite in left panel the app hanged after a fetch though

## Test methodology 

Manual use, just annotations

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by mstv — APPROVED** [review]
have not run

File: src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs [hunk 1]
```diff
@@ -35,6 +36,7 @@ public FormDashboardCategoryTitle(IEnumerable<string> existingCategories, string
         /// <summary>
         /// Gets the new category.
         /// </summary>
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
```
Comment (RussKie) [inline src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:39]: For all these cases I suggest adding `[Browsable(false)]` as well to hide the property from the PropertyGrid.

**Review by RussKie — APPROVED** [review]

File: src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs (hunk 1 above)
Comment (gerhardol) [reply src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:39]: So you mean [Browsable(false)] in addition to [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]?
[Browsable(false)] is set on most occurrences already.

Comment (RussKie) [reply src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:39]: > So you mean [Browsable(false)] in addition to [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]?

Yes, it'd be a good idea.

> [Browsable(false)] is set on most occurrences already.

There are some, yes.

Comment (gerhardol) [reply src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:40]: So with this being an issue in .net9: Should all  [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)] be reverted (and add ignore)?
Keep [Browsable(false)]?

Comment (gerhardol) [reply src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:40]: @RussKie I guess this has to be used, the error cannot be suppressed globally

Comment (RussKie) [reply src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/FormDashboardCategoryTitle.cs:40]: So, `<NoWarn>$(NoWarn);WFO1000</NoWarn>` doesn't work, does it? In this case, please open a new issue in dotnet/winforms.

/cc: @KlausLoeffelmann

**Comment by gerhardol** [comment]
Closing, as the check will be updated. Suppressing the warning in .net9.

**Pull request closed without merging**